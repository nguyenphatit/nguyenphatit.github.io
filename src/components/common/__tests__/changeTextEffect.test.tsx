import { fireEvent, render, screen } from '@testing-library/react';
import ChangeTextEffect from '../changeTextEffect';

describe('Change text effect ', () => {
    test('should render text and apply change text effect on hover', () => {
        const text = 'Hello world';
        render(<ChangeTextEffect text={text} />);
        const element = screen.getByTestId('text-hover-effect');
        expect(element).toHaveTextContent(text);

        fireEvent.mouseOver(element);
        expect(element).toHaveTextContent(text);
    });

    test('should match snapshot', () => {
        const { container } = render(<ChangeTextEffect text="Hello, world!" />);
        expect(container.firstChild).toMatchSnapshot();
    })
})
