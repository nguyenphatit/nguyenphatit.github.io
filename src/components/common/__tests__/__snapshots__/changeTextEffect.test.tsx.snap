// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change text effect  should match snapshot 1`] = `
<span
  class="text-hover-effect "
  data-testid="text-hover-effect"
>
  Hello, world!
</span>
`;
